File: java/version_00/pokemon/Pokemon.java
package pokemon;

public class Pokemon{
	public Pokemon(String n){
		this.name = n;
	}
	/**********************
			Getters
	**********************/

	public int getSR(){ return this.softResets; }
	public String getName(){ return this.name; }

	/**********************
			Setters
	**********************/

	public void setSR(int sr){ this.softResets = sr; }
	public void setName(String n){ this.name = n; }

	/**********************
			Attributes
	**********************/

	private String name;
	private int softResets;
}

File: c/version_00/pokemon/Pokemon.c
#include <stdio.h>
#include "Pokemon.h"

/**********************
		Getters
**********************/

int getSR(Pokemon p){
	return p.softResets;
}
char * getName(Pokemon *p){
	return p->name;
}

/**********************
		Setters
**********************/

void setSR(Pokemon *p, int sr){
	p->softResets = sr;
}

void setName(Pokemon *p){
	fflush(stdout);
	scanf("%s", p->name);
	getchar();
}